aten/src/ATen/native/mps/kernels/EmbeddingBag.metal

```diff
@@ -23,54 +23,72 @@ struct ReductionOpInit<EmbeddingBagMode::MAX, T> {
 template <EmbeddingBagMode M, typename T>
 struct ReductionOp {
   inline opmath_t<T> operator()(
-      T weight_val,
+      opmath_t<T> weight_val,
       opmath_t<T> out_val,
+      bool is_first) {
+    return weight_val + out_val;
+  }
+};
+
+template <typename T>
+struct ReductionOp<EmbeddingBagMode::MAX, T> {
+  inline opmath_t<T> operator()(
+      opmath_t<T> weight_val,
+      opmath_t<T> out_val,
+      bool is_first) {
+    return (is_first || weight_val > out_val) ? weight_val : out_val;
```

Comment:
This would also be a non-nan preserving,  but I guess it's the same behavior as CPU